### PR TITLE
basic supported build documentation

### PR DESCRIPTION
**Pull request by jvaltin**
Per #19, I've collected the responses and documented it.

### CODE REVIEW

**Comment by x1ddos** [comment]
Why not add this to the readme instead? Separate file is not as easily discoverable, especially when it isn't linked to from readme.

**Comment by jvaltin** [comment]
It's quite standard to have README, INSTALL, BUILD, and so on. I had thought that we could expand this into specific steps per platform and having it in another file keeps the original README succinct.

**Comment by x1ddos** [comment]
The text in the proposed change is mainly informational and I'd argue mostly relevant to README, the way it is right anyway.

> I had thought that we could expand this into specific steps per platform
Maybe it would make more sense to move this section into a separate file when it actually happens?

**Comment by jvaltin** [comment]
> Maybe it would make more sense to move this section into a separate file when it actually happens?

I've added those changes as well.

**Comment by jvaltin** [comment]
The build test failed for reasons that are unrelated to my changes.

**Comment by x1ddos** [comment]
> I've added those changes as well.

Much better. Thanks!

File: BUILD.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+The BitBox Wallet supports building on X86_64 machines. Building on 32bit
+systems is may be possible but it is explicitly unsupported. The following
+platforms should be viable for development, building, and use of the BitBit
```
[the remark this answers is not in the document]
Comment (jvaltin) [reply BUILD.md]: Good catch, thanks!

File: BUILD.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Prepare the MacOS system to have the build environment:
+`make osx-init`
+
+Build the QT frontend for MacOS:
+`make qt-osx`
+
+Build artifacts:
+* `frontends/qt/build/osx/BitBox.app`
+* `frontends/qt/build/osx/BitBox.dmg`
```
[the remark this answers is not in the document]
Comment (jvaltin) [reply BUILD.md]: Shouldn't we keep both as that is what we currently do? Also, why not sign both?

[the remark this answers is not in the document]
Comment (benma) [reply BUILD.md]: By sign I mean `codesign` by Apple, which works on .app folders. dmg would be a manual step after, if we decide to go dmg over zip.

You can keep both, but I will delete it very soon, so it does not matter ;)

[the remark this answers is not in the document]
Comment (jvaltin) [reply BUILD.md]: I've kept it for now as it accurately describes what is built, if it needs to be removed to merge, I'll do so. I think we should also document the release process and automate it such that the documentation is merely highlighting the specific Makefile targets.

File: BUILD.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+Initalize the Docker image:
+`make dockerinit`
+`make dockerdev`
+
+Inside of Docker, build the QT frontend:
+`make qt-linux`
+
+## Fedora
+
+Install [Docker for Fedora]
```
[the remark this answers is not in the document]
Comment (jvaltin) [reply BUILD.md]: Agreed, I've reworded it - though I've kept them as different sections with the hope that we might also break out the documentation for non-docker users.

**Pull request merged**